graphql/batch_test.go

```diff
@@ -0,0 +1,368 @@
+package graphql_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/samsarahq/thunder/graphql"
+	"github.com/samsarahq/thunder/graphql/schemabuilder"
+	"github.com/samsarahq/thunder/internal"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBatchFieldFuncExecution(t *testing.T) {
+	type Object struct {
+		Key string
+	}
+	tests := []struct {
+		Name                  string
+		GiveObjectFunc        interface{}
```

Comment:
idiomatic go (at least in the go compiler source) is to have all these fields be private, and to drop the give prefix (since it's implied by not having a want/desired prefix)